### PR TITLE
zsh: add programs.zsh.autosuggestions.strategy option

### PR DESCRIPTION
**Pull request by JohnRTitor**
### Description

zsh: add programs.zsh.autosuggestions.strategy option like in nixpkgs.
The option defaults to history.

### Checklist

- [x] Change is backwards compatible.

- [ ] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [ ] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

**Comment by JohnRTitor** [comment]
The eval is failing? Not sure why, though.

File: modules/programs/zsh.nix [hunk 1]
```diff
@@ -397,6 +397,22 @@ in
             Custom styles for autosuggestion highlighting. See
             {manpage}`zshzle(1)` for syntax.
           '';
```
Comment (teto) [inline modules/programs/zsh.nix:399]: the previous option is not closed, add a `};`

Comment (JohnRTitor) [reply modules/programs/zsh.nix:399]: Done. Hopefully eval succeeds now.

Comment (JohnRTitor) [reply modules/programs/zsh.nix:399]: Nope. It does not.

**Comment by JohnRTitor** [comment]
@rycee I think this is ready for review. Please check. Thanks

**Comment by JohnRTitor** [comment]
@rycee hi, if you are free to review this, could you check? Just a friendly ping :)

**Comment by JohnRTitor** [comment]
Ping

**Comment by Mic92** [comment]
Also I have technically merge rights here because I am administrating nix-community, I don't think I am a proper home-manager maintainer.

**Comment by JohnRTitor** [comment]
The thing is, home manager itself has very few committers and reviewers, unlike nixpkgs. So getting a patch merged takes a lot of time.

**Pull request merged**